### ISSUE TITLE
Ensure running with a config file outside ~/.sopel works as expected

### ISSUE THREAD

**Issue by embolalia** [sopel-irc/sopel#946]
- [x] Config wizard should not create ~/.sopel if the specified config is not in that directory
- [x] Existing code should be checked to ensure nothing else assumes things in ~/.sopel
- [x] How to specify a config outside of ~/.sopel should be documented

**Comment by Exirel** [comment]
I should have a look while working on #1429

**Comment by dgw** [comment]
This issue gets a specific callout in [irctest](https://github.com/ProgVal/irctest)'s readme, and I think we would be well served to solve whatever remains of this issue for the Sopel 7.0 release. I know we've already done some homedir-related stuff, and if my hunch is correct we're already most of the way to closing this.

**Comment by dgw** [comment]
Preliminary list of related PRs: #1404 (merged), #1574 (awaiting final review from yours truly), & #1598 (awaiting final tweaks from @Exirel). About the only unaddressed reference to `~/.sopel` I could find in my brief survey was a hard-coded reference in `test_tools` that should be updated to use the `config.DEFAULT_HOMEDIR` constant added way back in #1429. https://github.com/sopel-irc/sopel/blob/3404aa29f6395952d5f984c5fa2183a3cf6936a4/sopel/test_tools.py#L97

**Comment by Exirel** [comment]
I need to double check on my end, but as @dgw said, it's almost done yes.

**Comment by Exirel** [comment]
Fun fact, the rework of test tools in #1731 make the hard-coded home_dir unnecessary, because we can use a real `Sopel` instance instead of the `MockSopel`. So it's almost done: all that is required now is to 1) merge #1731 then 2) use the new tools in every tests.

**Comment by dgw** [comment]
> So it's almost done: all that is required now is to 1) merge #1731 then 2) use the new tools in every tests.

FWIW, `test_tools` still has the same old hard-coded `home_dir` value. @Exirel is tweaking that even worth blocking close of this issue for? I feel like it isn't.

**Comment by Exirel** [comment]
I'm working on that last part with #1781 and I'm probably going to remove both `MockSopel` and `MockSopelWrapper` (and also `MockConfig`).

**Comment by Exirel** [comment]
> is tweaking that even worth blocking close of this issue for? I feel like it isn't.

Agree, we can totally close that issue once #1781 is merged, because Sopel won't use the deprecated code itself.

**Comment by dgw** [comment]
Checked off the "existing code" task now that #1781 is merged. We should be golden… FINALLY! 🎉
